airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/tests/test_core.py

```diff
@@ -1175,3 +1179,170 @@ async def test_streams_define_primary_key(
+        not_required_heading = (
+            f"Set up the {connector_name} connector in Airbyte",
+            "For Airbyte Cloud:",
+            "For Airbyte Open Source:",
+            self.CONNECTOR_SPECIFIC_HEADINGS,
+            "Performance considerations",
+            "Data type map",
+            "Troubleshooting",
+            "Tutorials",
+        )
+        return all_headings, not_required_heading
+
+    def _headings_description(self, connector_name: str) -> dict[str:Path]:
+        """
+        Headings with path to file with template description
+        """
+        descriptions_paths = {
+            connector_name: Path(__file__).parent / "doc_templates/source.txt",
+            "For Airbyte Cloud:": Path(__file__).parent / "doc_templates/for_airbyte_cloud.txt",
+            "For Airbyte Open Source:": Path(__file__).parent / "doc_templates/for_airbyte_open_source.txt",
+            "Supported sync modes": Path(__file__).parent / "doc_templates/supported_sync_modes.txt",
+            "Tutorials": Path(__file__).parent / "doc_templates/tutorials.txt",
+        }
+        return descriptions_paths
+
+    def test_prerequisites_content(self, actual_connector_spec: ConnectorSpecification, connector_documentation: str, docs_path: str):
+        node = docs_utils.documentation_node(connector_documentation)
+        header_line_map = {docs_utils.header_name(n): n.map[1] for n in node if n.type == self.HEADING}
+        headings = tuple(header_line_map.keys())
+
+        if not header_line_map.get(self.PREREQUISITES):
+            pytest.fail(f"Documentation does not have {self.PREREQUISITES} section.")
+
+        prereq_start_line = header_line_map[self.PREREQUISITES]
+        if headings.index(self.PREREQUISITES) + 1 == len(headings):
+            prereq_end_line = -1
+        else:
+            prereq_end_line = header_line_map[headings[headings.index(self.PREREQUISITES) + 1]]
+
+        with open(docs_path, "r") as docs_file:
+            prereq_content_lines = docs_file.readlines()[prereq_start_line:prereq_end_line]
+            prereq_content = "".join(prereq_content_lines).lower()
+            required_titles, has_credentials = docs_utils.required_titles_from_spec(actual_connector_spec.connectionSpecification)
+
+            for title in required_titles:
+                assert title in prereq_content, f"Required '{title}' field is not in {self.PREREQUISITES} section."
+
+            if has_credentials:
+                credentials_validation = [k in prereq_content for k in self.CREDENTIALS_KEYWORDS]
+                assert True in credentials_validation, f"Required 'credentials' field is not in {self.PREREQUISITES} section."
+
+    def test_docs_structure(self, connector_documentation: str, connector_metadata: dict):
+        node = docs_utils.documentation_node(connector_documentation)
+        heading_names = tuple([docs_utils.remove_step_from_heading(docs_utils.header_name(n)) for n in node if n.type == self.HEADING])
```

Comment:
just to confirm, this will get headers at nested levels too right?

Reply:
no, it gets only top-level headers

Reply:
Looking at the basic template and some existing docs, the Step 1, Step 2...etc are all on the second level after `Setup guide`. So if we wanted to test that we have to check nested right?

But I also did a glance over a bunch of certified connector docs and it looks like only a few follow this `Step 1: Set up <source>` and `Step 2: Set up the <source> connector in Airbyte`. Even though this test is opt-in, I think we should just check that `Setup guide` is at the top level as that is the most important part. Whether it follows the sub-header step pattern I don't think is as important

Reply:
yes, we are validating step 1 and step 2, it just removes Step 1/2 heading name. So `Step 1: Set up <connector name>` str becomes `Set up <connector name>` as in required header. The reason why I removed it that we don't know the step number of this header, `Set up <connector name>`  can be a step 1 or step 2.